### PR TITLE
Avoid deserializing image on the edge-endpoint

### PR DESCRIPTION
**Pull request by tyler-romero**
This will hopefully provide a little speedup.

### CODE REVIEW

File: app/api/routes/image_queries.py [hunk 1]
```diff
@@ -22,32 +20,17 @@
...
-        # Image.open() does not fully process the image data. It's possible for Image.open()
-        # to succeed but then fail when the image data is actually being processed.
-        # To ensure that the image can be fully processed, we call img.load() to force loading
-        # the entire image. If this fails, we know that the image is invalid.
-        image.load()
-    except IOError as ex:
-        logger.error("Failed to load image", exc_info=True)
-        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail="Invalid input image") from ex
-    return image
+async def validate_image_bytes(request: Request, content_type: str = Depends(validate_content_type)) -> bytes:
```
Comment (tyler-romero) [inline app/api/routes/image_queries.py:31]: note: Dependency injection in fastapi forms a DAG. Nodes of the DAG do not need to be recomputed. So validate_content_type is computed once per request and the result is used in multiple places. In this case, it is to perform validation before attempting to return the image. Below, it is actually to get the content-type.

File: app/api/routes/image_queries.py [hunk 2]
```diff
@@ -22,32 +20,17 @@
...
     if not request.headers.get("Content-Type", "").startswith("image/"):
         raise HTTPException(
             status_code=status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, detail="Request body must be image bytes"
         )
+    return request.headers.get("Content-Type", "")
 
-    image_bytes = await request.body()
-    try:
-        # Attempt to open the image
-        image = Image.open(BytesIO(image_bytes))
```
Comment (tyler-romero) [inline app/api/routes/image_queries.py:40]: Potential speedup from avoiding this

File: app/core/edge_inference.py [hunk 3]
```diff
@@ -29,16 +27,11 @@ def is_edge_inference_ready(inference_client_url: str) -> bool:
         return False
 
 
-def submit_image_for_inference(inference_client_url: str, image: Image.Image) -> dict:
+def submit_image_for_inference(inference_client_url: str, image_bytes: bytes, content_type: str) -> dict:
     inference_url = f"http://{inference_client_url}/infer"
-
-    # Convert the PIL image to a WebP byte stream
-    byte_arr = BytesIO()
-    image.save(byte_arr, format="WEBP")
```
Comment (tyler-romero) [inline app/core/edge_inference.py:37]: Potential speedup by avoiding this

**Review by CoreyEWood — APPROVED** [review]
Looks good, is it possible to test this before merging to make sure everything works?

**Comment by tyler-romero** [comment]
Just stood it all up and ensured it works!

**Pull request merged**